### ISSUE TITLE
Spitzer lightcurves

### ISSUE THREAD

**Issue by gully** [KeplerGO/lightkurve#603]
Spitzer acquired precision time series photometry in point-and-stare conditions for targeted science cases like transiting exoplanets or weather on brown dwarfs.  Those data are housed as single per-cadence fits images, with of order ten second cadence, for durations of up to several days.  

I have recently demo-ed that it's possible to read in these fits files to a `lightkurve` TPF using a modified TPF-factory, and some hacks of fits header keywords.  I got tpf.interact(), and periodogram() to work.  PLDcorrector yielded unexpected results that I'm still working to understand.

In short, it's clearly possible to use lightkurve to analyze Spitzer data.  The question becomes how much to alter the relative base classes, and how we can improve the structure of the code to make it more robust and reduce code duplication between Kepler and TESS.

**Comment by gully** [comment]
Things to do:
- [ ] Check the `tpf-ext{}-header.txt` fits header template files for their applicability to generic TPFs
- [x] Adapt `self.mission` property
- [x] Apply `self.mission` for control flow
- [ ] Spot check `wcs` attribute robustness

**Comment by gully** [comment]
Question:
Should `LightCurve` objects support centroids?

**Comment by barentsen** [comment]
As of Lightcurve v2.x, the LightCurve object has been turned into a sub-class of AstroPy TimeSeries.  This now allows a LightCurve object to store Spitzer centroids or any other non-Kepler/Tess data in a generic way, either as extra columns or in the flexible `meta` dictionary.

I believe that resolves this issue!
